### PR TITLE
Add support for getting static discovery info from a raw string [10841]

### PR DESCRIPTION
**Pull request by richiware**
This PR adds the feature of support a static discovery XML read from a string.
An example:

```c++
participant_qos_.wire_protocol().builtin.discovery_config.static_edp_xml_content("data://<staticdiscovery>...</staticdiscovery>");
```

Related documentation on eProsima/Fast-DDS-docs#254.

### CODE REVIEW

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please test this

**Comment by richiware** [comment]
@richiprosima Please check style

**Review by EduPonz — CHANGES_REQUESTED** [review]
I think we should create and API so normal users can benefit from the feature as well. Furthermore, I'd like a related documentation PR. Kudos for the tests! 🚀

File: include/fastdds/rtps/attributes/RTPSParticipantAttributes.h [hunk 1]
```diff
@@ -301,25 +301,50 @@ class DiscoverySettings
...
-        m_staticEndpointXMLFilename = std::string(str);
+        static_edp_xml_content_ = "file://" + std::string(str);
+    }
+
+    /**
+     * Set the static endpoint XML content.
+     * @param str URI specifying the static endpoint XML content.
+     * The string could contain a filename (file://) or the XML content directly (data://).
+     */
+    void static_edp_xml_content(
```
Comment (EduPonz) [inline include/fastdds/rtps/attributes/RTPSParticipantAttributes.h:327]: Knowing that you were not convinced with the name, I propose an alternative. Mind that you'd need to propagate the change
```suggestion
    void static_edp_xml_config(
```

File: include/fastdds/rtps/attributes/RTPSParticipantAttributes.h [hunk 2]
```diff
@@ -301,25 +301,50 @@ class DiscoverySettings
      * Get the static endpoint XML filename
      * @return Static endpoint XML filename
      */
+    FASTRTPS_DEPRECATED("Use static_edp_xml_content()")
     const char* getStaticEndpointXMLFilename() const
     {
-        return m_staticEndpointXMLFilename.c_str();
+        return static_edp_xml_content_.c_str();
```
Comment (EduPonz) [inline include/fastdds/rtps/attributes/RTPSParticipantAttributes.h:307]: I'd have the deprecated function calling the new one

File: test/blackbox/common/BlackboxTestsDiscovery.cpp [hunk 3]
```diff
@@ -223,6 +223,92 @@ TEST(Discovery, StaticDiscovery)
     reader.block_for_all();
 }
 
+TEST(Discovery, StaticDiscoveryFromString)
```
Comment (EduPonz) [inline test/blackbox/common/BlackboxTestsDiscovery.cpp:226]: I'd like a description of the general purpose of the tests followed by a step by step guide on the actions taken

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM with green CI 🚀

File: test/blackbox/common/BlackboxTestsDiscovery.cpp [hunk 4]
```diff
@@ -225,6 +225,30 @@ TEST(Discovery, StaticDiscovery)
 
 /*!
  * Test Static EDP discovery configured via a XML content in a raw string.
+ *
+ * Currently Fast DDS API supports configure Static EDP discovery in two ways: setting the file containing the XML
+ * configuration or passing directly the XML content. This tests tests the second way.
```
Comment (EduPonz) [inline test/blackbox/common/BlackboxTestsDiscovery.cpp:230]:
```suggestion
 * configuration or passing directly the XML content. This test tests the second way.
```

File: test/blackbox/common/BlackboxTestsDiscovery.cpp [hunk 5]
```diff
@@ -225,6 +225,30 @@ TEST(Discovery, StaticDiscovery)
...
+ *
+ * 2. Initialize writer.
+ *
+ * 3. Configure a reader. Static EDP Discovery is enable and XML configuration is passed directly using
+ * static_edp_xml_config() API funcion.
+ *
+ * 4. Initialize writer.
+ *
+ * 5. Wait both entities discover between them. If the Static EDP Discovery was configured correctly, they should
+ * discovery.
```
Comment (EduPonz) [inline test/blackbox/common/BlackboxTestsDiscovery.cpp:245]:
```suggestion
 * discover each other.
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]
LGTM with green CI 🤖

**Pull request merged**